### PR TITLE
refactor(framework): EO-1095 Framework Elements Wells

### PR DESCRIPTION
**Pull request by manojdas2212**
JIRA: https://jira.rax.io/browse/EO-1095

### LGTMs
- [x] Dev LGTM
- [x] Dev+Vidyo LGTM
- [x] Design LGTM

Before
======

Layout > Wells
-------------------------------
screen shot 2016-08-17 at 9 39 46 pm

After
======
Elements > Wells
-------------------------------
screen shot 2016-08-17 at 9 40 31 pm

Full Page 
======

screen shot 2016-08-20 at 11 24 58 pm

screen shot 2016-08-17 at 9 40 58 pm

View All Modules
======
filter by "well" 
-------------------------------

screen shot 2016-08-17 at 9 57 38 pm

### CODE REVIEW

**Comment by CITguy** [comment]
@manojdas2212: This PR does not meet the acceptance criteria for the JIRA story. Please update.

**Comment by manojdas2212** [comment]
@CITguy Yes Ryan, this is initial commit as you mentioned in JIRA story so i will update more in this PR later

**Comment by manojdas2212** [comment]
@CITguy Any "API DOCS" associated with "Wells" please confirm me

**Comment by CITguy** [comment]
Wells do not have any javascript associated with them, so there should not be any API docs.

**Comment by manojdas2212** [comment]
@CITguy I have updated the changes please take a look

File: src/elements/Wells/docs/wells.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+   To create a well-effect, you can use the <code>.well</code>
+   class &mdash; a box with significant padding and light gray background.
+   You might use Page wells to communicate why data is not present for the
+   user, usually if data can only be seen with permissions the user has not
+   been granted. In cases where a user is able to add data without a
+   permissions change, a button can be displayed to prompt the user
+   to add data.
+</p>
+<rx-example name="wells.simple"></rx-example>
+<h3>Modal Wells</h3>
```
Comment (khaziwallis) [inline src/elements/Wells/docs/wells.html]: @manojdas2212 one line after one example.
So shift <h3> Modal Wells</h3> to line 13.

File: src/elements/Wells/docs/examples/wellsmodal.simple.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<rx-modal-action classes="button" template-url="modal-well.html">Open a modal</rx-modal-action>
```
Comment (khaziwallis) [inline src/elements/Wells/docs/examples/wellsmodal.simple.html]: @manojdas2212 i guess we need to change the name to `wells.modal.html` for consistency with other `element`

File: src/elements/Wells/docs/wells.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<p>
+   Typically you can use Modal wells in the context of modals, to provide a
+   summary of data entered or decisions made by the user in a form, to provide
+   a clearer picture to the user of the changes they are about to submit.
+</p>
+<rx-example name="wellsmodal.simple"></rx-example>
```
Comment (khaziwallis) [inline src/elements/Wells/docs/wells.html]: @manojdas2212 i need to be wells.modal, which corresponds to above comment.

File: src/elements/Wells/docs/examples/wellsmodal.simple.html (hunk 2 above)
Comment (CITguy) [inline src/elements/Wells/docs/examples/wellsmodal.simple.html]: The `template-url="modal-well.html"` is fine, but the file itself should probably be renamed to `wells.modal.html`.

**Comment by CITguy** [comment]
Removed QE LGTM (no javascript to test), and updated labels. 

I'm in agreement with existing feedback comments.

**Comment by CITguy** [comment]
@manojdas2212: To remove the "View API" button at the top of the page, you'll need to add `"hasApi": false` to the `*.meta.json` file.

File: src/elements/Wells/Well.meta.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+    "displayName": "Wells",
+    "stability": "stable",
```
Comment (khaziwallis) [inline src/elements/Wells/Well.meta.json]: add this line
`hasApi": false,`

**Comment by manojdas2212** [comment]
@khaziwallis @CITguy Addressed your comments please take a look

File: src/elements/Wells/docs/wells.html [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<h3 class="clear">Page Wells</h3>
```
Comment (100stacks) [inline src/elements/Wells/docs/wells.html]: @manojdas2212, please change the file name to `wells.docs.html`.

File: src/elements/Wells/docs/examples/wells.docs.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<div class="well centered">
```
Comment (100stacks) [inline src/elements/Wells/docs/examples/wells.docs.html]: @manojdas2212, please rename file to `wells.permissions.html`.

File: src/elements/Wells/docs/wells.html [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+<p>
+   To create a well-effect, you can use the <code>.well</code>
+   class &mdash; a box with significant padding and light gray background.
+   You might use Page wells to communicate why data is not present for the
+   user, usually if data can only be seen with permissions the user has not
+   been granted. In cases where a user is able to add data without a
+   permissions change, a button can be displayed to prompt the user
+   to add data.
+</p>
+<rx-example name="wells.docs"></rx-example>
```
Comment (100stacks) [inline src/elements/Wells/docs/wells.html]: @manojdas2212, please rename to `wells.permissions`

File: src/elements/Wells/Wells.meta.json [hunk 8]
```diff
@@ -0,0 +1,8 @@
+{
+    "displayName": "Wells",
+    "stability": "stable",
+    "hasApi": false,
```
Comment (100stacks) [inline src/elements/Wells/Wells.meta.json]: @manojdas2212, please add:

> "description":"Provides usage of wells in different context.",

**Comment by manojdas2212** [comment]
@CITguy @100stacks I have updated the changes please take a look

**Comment by CITguy** [comment]
Dev LGTM

**Comment by 100stacks** [comment]
LGTM

**Comment by ElementE** [comment]
(might be an artifact of the screenshots - but the status is different in the list than on the page itself)
Other than that - Design LGTM

**Comment by CITguy** [comment]
Old screenshot, yes. I can verify that the stability is as expected.

screen shot 2016-08-23 at 3 03 50 pm
screen shot 2016-08-23 at 3 03 44 pm

**Pull request merged**